android/expoview/src/main/java/versioned/host/exp/exponent/modules/api/notifications/NotificationsModule.java

```diff
@@ -356,14 +365,121 @@ public void cancelScheduledNotificationAsync(final int notificationId, final Pro
     }
   }
 
+  @ReactMethod
+  public void cancelScheduledNotificationWithStringIdAsync(final String id, final Promise promise) {
+    try {
+      SchedulersManagerProxy.getInstance(getReactApplicationContext()
+          .getApplicationContext())
+          .removeScheduler(id);
+      promise.resolve(null);
+    } catch (Exception e) {
+      promise.reject(e);
+    }
+  }
+
   @ReactMethod
   public void cancelAllScheduledNotificationsAsync(final Promise promise) {
     try {
       ExponentNotificationManager manager = new ExponentNotificationManager(getReactApplicationContext());
       manager.cancelAllScheduled(mManifest.getString(ExponentManifest.MANIFEST_ID_KEY));
+
+      String experienceId = mManifest.optString(ExponentManifest.MANIFEST_ID_KEY, null);
+
+      SchedulersManagerProxy
+          .getInstance(getReactApplicationContext().getApplicationContext())
+          .removeAll(experienceId);
+
       promise.resolve(null);
     } catch (Exception e) {
       promise.reject(e);
     }
   }
+
+  @ReactMethod
+  public void scheduleNotificationWithTimer(final ReadableMap data, final ReadableMap optionsMap, final Promise promise) {
+    HashMap<String, Object> options = optionsMap.toHashMap();
+    int notificationId = Math.abs( new Random().nextInt() );
```

Comment:
Can't this `id` be set by the database? What if we use an int that is already present in the database?

Reply:
If we assume that we currently have 100 notifications then the probability that two of them have the same id is around e ^ (- 50*99/1e9) ~ 0.999995050012251229785462515496328404548599905391691594227...